plugins/backend/backstage-plugin-argo-cd-backend/src/refs/argocdService.ref.ts

```diff
@@ -1,6 +1,22 @@
-import { createServiceRef } from '@backstage/backend-plugin-api';
+import { coreServices, createServiceFactory, createServiceRef } from '@backstage/backend-plugin-api';
+import { ArgoService } from '../service/argocd.service';
 import { ArgoServiceApi } from '../service/types';
 
 export const argocdServiceRef = createServiceRef<ArgoServiceApi>({
   id: 'argocd-service-backend',
+  defaultFactory: async service =>
+    createServiceFactory({
+      service,
+      deps: {
+        config: coreServices.rootConfig,
+        logger: coreServices.logger,
+      },
+      async factory({ config, logger }) {
+        const argoUserName =
+          config.getOptionalString('argocd.username') ?? 'argocdUsername';
+        const argoPassword =
+          config.getOptionalString('argocd.password') ?? 'argocdPassword';
+        return new ArgoService(argoUserName, argoPassword, config, logger);
```

Comment:
This API is unexpected and seems overly complex. You have access to the entirety of the `config` so it seems odd to pull some pieces of the config out and also pass those along with the entire config. I just copied this from the service factory as is, but I think this could be cleaned up. I'll do it in a follow up pull request.